hoomd/hpmc/external/wall.py

```diff
@@ -1,525 +1,111 @@
+    See Also:
+        `hoomd.wall`
 
-        You may want to set this independently so that you can cleverly control whether or not the walls actually scale in case you manually resize your simulation box.
-        The walls scale automatically when they get the signal that the global box, associated with the system definition, has scaled. They do so, however, with a scale factor associated with
-        the ratio of the volume of the global box to the volume of the box that the walls class is currently storing. (After the scaling the box that the walls class is currently storing is updated appropriately.)
-        If you want to change the simulation box WITHOUT scaling the walls, then, you must first update the simulation box that the walls class is storing, THEN update the global box associated with the system definition.
+    Example::
 
-        Example::
+        mc = hoomd.hpmc.integrate.Sphere()
+        walls = [hoomd.wall.Sphere(radius=4.0)]
+        wall_potential = hoomd.hpmc.external.wall.WallPotential(walls)
+        mc.external_potential = wall_potential
 
-            init_box = hoomd.data.boxdim(L=10, dimensions=3);
-            snap = hoomd.data.make_snapshot(N=1, box=init_box, particle_types=['A']);
-            system = hoomd.init.read_snapshot(snap);
-            system.particles[0].position = [0,0,0];
-            system.particles[0].type = 'A';
-            mc = hpmc.integrate.sphere(seed = 415236);
-            mc.shape_param.set('A', diameter = 2.0);
-            ext_wall = hpmc.compute.wall(mc);
-            ext_wall.add_sphere_wall(radius = 3.0, origin = [0, 0, 0], inside = True);
-            ext_wall.set_curr_box(Lx=2.0*init_box.Lx, Ly=2.0*init_box.Ly, Lz=2.0*init_box.Lz, xy=init_box.xy, xz=init_box.xz, yz=init_box.yz);
-            system.sysdef.getParticleData().setGlobalBox(ext_wall.get_curr_box()._getBoxDim())
+    """
 
-        """
-        # much of this is from hoomd's update.py box_resize class
-        if all((Lx is None, Ly is None, Lz is None, xy is None, xz is None,
-                yz is None)):
-            hoomd.context.current.device.cpp_msg.warning(
-                "compute.wall.set_curr_box: Ignoring request to set the wall's \
-                        box without parameters\n")
+    def __init__(self, walls):
+        self._walls = hoomd.wall._WallsMetaList(walls, _to_hpmc_cpp_wall)
+
+    def _check_valid_shape_wall_pairs(self, walls, integrator):
+        supported_shape_wall_overlap_checks = {
+            hoomd.hpmc.integrate.Sphere: [
+                hoomd.wall.Sphere, hoomd.wall.Cylinder, hoomd.wall.Plane
+            ],
+            hoomd.hpmc.integrate.ConvexPolyhedron: [
+                hoomd.wall.Sphere, hoomd.wall.Cylinder, hoomd.wall.Plane
+            ],
+            hoomd.hpmc.integrate.ConvexSpheropolyhedron: [
+                hoomd.wall.Sphere, hoomd.wall.Plane
+            ]
+        }
+        integrator_type = type(integrator)
+        for wt in [type(w) for w in walls]:
```

Comment:
```suggestion
        for wt in (type(w) for w in walls):
```
Use generator for for-loops.